### PR TITLE
CSO search

### PR DESCRIPTION
**Pull request by TayGov**
# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/libs/efiling-cso-client/src/main/java/ca/bc/gov/open/jag/efilingcsoclient/CsoSearchServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    private final CSOSearchSoap csoSearchSoap;
+
+    public CsoSearchServiceImpl(CSOSearchSoap csoSearchSoap) {
+        this.csoSearchSoap = csoSearchSoap;
+    }
+
+    @Override
+    public boolean caseNumberExists(String caseNumber) {
+
+        logger.info("search for case number request received");
```
Comment (akroon3r) [inline src/backend/libs/efiling-cso-client/src/main/java/ca/bc/gov/open/jag/efilingcsoclient/CsoSearchServiceImpl.java:23]: do we want to log the number?

Comment (TayGov) [reply src/backend/libs/efiling-cso-client/src/main/java/ca/bc/gov/open/jag/efilingcsoclient/CsoSearchServiceImpl.java:23]: Not sure how pii that is. I think we are safe not at this point

**Review by akroon3r — COMMENTED** [review]
One question

File: src/backend/libs/efiling-cso-client/src/main/java/ca/bc/gov/open/jag/efilingcsoclient/CsoSearchServiceImpl.java (hunk 1 above)
Comment (akroon3r) [reply src/backend/libs/efiling-cso-client/src/main/java/ca/bc/gov/open/jag/efilingcsoclient/CsoSearchServiceImpl.java:23]: 🐦

**Review by akroon3r — APPROVED** [review]

**Pull request merged**